src/modsim2/similarity/metrics.py

```diff
@@ -0,0 +1,167 @@
+
+        return matrix_A, matrix_B, N_A, N_B
+
+    def calculate_metric(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+        labels_A: np.ndarray,
+        labels_B: np.ndarray,
+        embedding_name: str,
+        kernel_name: str,
+    ) -> float:
+        """
+        Computes maximum mean discrepancy (MMD) for A and B. Given a kernel
+        embedding k(X,X'), computes
+
+            MMD^2 = 1/(N_A^2)*sum(k(A,A)) + 1/(N_B^2)*sum(k(B,B)) -
+                        2/(N_A * N_B)*sum(k(A,B))
+
+        Note that A and B must be matrices for the MMD. Currently mmd()
+        will reshape arbitarily size np.ndarrays to matrices. However,
+        other options include feature embeddings which will be explored in
+        the future.
+
+        Args:
+            array_A: The first image dataset
+            array_B: The second image dataset
+            embedding_name: What feature embeddings, if any, to use for the
+                            input arrays
+            kernel_name: An appropriate kernel embedding. See kernel_dict
+                        for choices
+
+        Returns:
+            float: The maximum mean discrepancy between A and B
+        """
+
+        matrix_A, matrix_B, N_A, N_B = self._pre_process_data(
+            data_A, data_B, embedding_name
+        )
+
+        # Extract kernel callable
+        kernel_fn = self.MMD_KERNEL_DICT[kernel_name]
+
+        # Compute MMD components
+        logging.info(
+            f"Computing {N_A} by {N_A}, {N_B} by {N_B}, and {N_A} by {N_B} kernels. "
+            "This may take some time!"
+        )
+        kernel_AA = kernel_fn(X=matrix_A)
+        kernel_BB = kernel_fn(X=matrix_B)
+        kernel_AB = kernel_fn(X=matrix_A, Y=matrix_B)
+
+        # Compute MMD
+        mmd = (
+            np.sum(kernel_AA) / (N_A**2)
+            + np.sum(kernel_BB) / (N_B**2)
+            - 2 * np.sum(kernel_AB) / (N_A * N_B)
+        )
+
+        # Return
+        return mmd
+
+
+class OTDD(DistanceMetric):
+    def __init__(self, seed: int = 42):
+        super().__init__(seed)
+
+    def _pre_process_data(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+        targets_A: np.ndarray,
+        targets_B: np.ndarray,
+    ):
+        dataset_A = TensorDataset(torch.tensor(data_A), torch.tensor(targets_A))
+        dataset_B = TensorDataset(torch.tensor(data_B), torch.tensor(targets_B))
+        dataset_A.classes = torch.sort(torch.unique(torch.tensor(targets_A))).values
+        dataset_B.classes = torch.sort(torch.unique(torch.tensor(targets_B))).values
+
+        return dataset_A, dataset_B
+
+    def calculate_metric(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+        labels_A: np.ndarray,
+        labels_B: np.ndarray,
+        max_samples: int,
+        device: str,
+        **kwargs,
+    ) -> float:
+        """
+        Calculates the optimal transport distance between datasets
```

Comment:
Very very minor comment, but for clarity should be `optimal transport dataset distance`